pkg/pillar/zboot/zboot.go

```diff
@@ -64,8 +64,24 @@ func Poweroff(log *base.LogObject) {
 
 // If log is nil there is no logging
 func execWithRetry(log *base.LogObject, command string, args ...string) ([]byte, error) {
+	retrySignals := map[syscall.Signal]struct{}{
+		syscall.SIGUSR1: {},
+		syscall.SIGUSR2: {},
```

Comment:
How did we decide these two signals are enough?

Reply:
By guessing ;-)

Do you have other signals in mind?

Afaik, these are the only signals that zedbox receives for some kind of IPC outside of wanting zedbox to terminate (and then child processes should be terminated as well).

Reply:
Those are the only signals we generate AFAIK. I'm assuming SIGPROF, if used by the go runtime for profiling, is handled internally by the runtime without any errors returned by exec. Might be good to check that.

Reply:
But in general, does it make sense to use the whitelisting approach here? Would handling all the signals that way be a disadvantage? In the future, we can add new signal handling to the system and forget about this place...

Reply:
> Would handling all the signals that way be a disadvantage? 

F.e. I think we should not handle `SIGSEGV` this way, because that can just end up in an endless loop.

Reply:
Maybe we can use blacklisting in this case?

Reply:
Blindly restarting a syscall which was interrupted by a signal (aynch or sync) seems risky; makes sense to do it for the signals we know are used as part of normal operation and not others.

Reply:
Okay, that makes sense. Could we add some kind of reminder to the place in the code where we register the handlers? That way, the next person adding a handler will see that there is one more extra place to be fixed.

Reply:
@OhmSpectator currently they're in `agentlog.go`, but I doubt that f.e. a handler for `SIGTERM` would end up there, too. Do you mean something like https://github.com/christoph-zededa/eve/commit/26df4981707260a10a0ef5e78ce15eefd147c858 ?

@eriknordmark 
>  if used by the go runtime for profiling, is handled internally by the runtime without any errors returned by exec
I checked with `strace`; yes it is using `SIGPROF` and it also installs a signal handler:
```
rt_sigaction(SIGPROF, {sa_handler=0x996920, sa_mask=~[RTMIN RT_1 RT_2], sa_flags=SA_RESTORER|SA_ONSTACK|SA_RESTART|SA_SIGINFO, sa_restorer=0x7fe613a3eaca}, NULL, 8)
```
(and `clone`s are done with `CLONE_SIGHAND`)

but isn't it always sending the signal to it's own pid anyways (by design of the syscall)?
```
timer_create(CLOCK_THREAD_CPUTIME_ID, {sigev_signo=SIGPROF, sigev_notify=SIGEV_THREAD_ID, sigev_notify_thread_id=30794}, [0]) = 0
```